library/Zend/Http/Client/Adapter/Curl.php

```diff
@@ -228,10 +228,12 @@ public function connect($host, $port = 80, $secure = false)
             curl_setopt($this->_curl, CURLOPT_CONNECTTIMEOUT, $this->_config['timeout']);
         }
 
-        if (defined('CURLOPT_TIMEOUT_MS')) {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['timeout'] * 1000);
-        } else {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['timeout']);
+        if (array_key_exists('request_timeout', $this->_config)) {
+          if (defined('CURLOPT_TIMEOUT_MS')) {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['request_timeout'] * 1000);
+          } else {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['request_timeout']);
+          }
```

Comment:
Also, I think if we're going to take either approach, the other conditionals that do similar checks for features should be done in the same way (there's a similar block immediately above for the CONNECTTIMEOUT constant variations).

I can do that on merge.